src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -99,7 +103,9 @@ public static long copyToFileAndClose(InputStream is, File file, long timeout) t
   public static long copyAndClose(InputStream is, OutputStream os) throws IOException
   {
     try {
-      return ByteStreams.copy(is, os);
+      final long retval = ByteStreams.copy(is, os);
+      os.flush();
```

Comment:
the reason the exception is getting swallowed is because we use CloseQuietly.close(os). Instead of flushing we should simply not swallow exceptions when closing the output or input stream.